src/murmur/GRPCall.h

```diff
@@ -0,0 +1,1248 @@
+			/// the old method. At the moment, only the Derived::impl() method is
+			/// called here. To match the old interface, those functions can throw
+			/// `grpc::Status` like it was an exception to indicate an error.
+			///
+			/// If an `grpc::Status` is caught, if the service was not cancelled, it is
+			/// used to call error() which will notify gRPC of the failure and close
+			/// the connection.
+			///
+			/// \param fn a *MoveConstructable* function with signature `void()` to be
+			/// run in the main murmur event loop. Ownership will be taken of the function.
+			///
+			template<typename Functor>
+			void launchInEventLoop(Functor&& fn) {
+				static_assert(std::is_same<ct::function_type_t<Functor>, void()>::value,
+							"launchInEventLoop(Fn) expects void() for fn");
+
+#if defined(BOOST_NO_CXX14_INITIALIZED_LAMBDA_CAPTURES)
+				boost::fibers::fiber f(
+					std::bind([](decltype(m_this) ptr, Functor& func) {
+						try {
+							func();
+						} catch (::grpc::Status &ex) {
+							if (!ptr->isCancelled()) {
+								ptr->error(ex);
+							}
+						}
+					}, std::move(this->getSharedPtr()), std::forward<Functor>(fn)));
+#else
+				boost::fibers::fiber f([ptr = std::move(this->getSharedPtr()), func = std::forward<Functor>(fn)](){
+					try {
+						func();
+					} catch (::grpc::Status &ex) {
+						if (!ptr->isCancelled()) {
+							ptr->error(ex);
+						}
+					}
+				});
+#endif
+				auto& props( f.properties<MurmurRPC::Scheduler::grpc_props>() );
+				props.run_in_main(true);
+				f.detach();
+			}
+
+			/// \brief function called by gRPC when a new service is requested.
+			///
+			/// Creates a new identical
+			/// RPCall to handle the next new service request. It then acts somewhat differently
+			/// depending on the type of service.
+			///
+			/// For unary services, the incoming message has already been read in. It then calls
+			/// Derived::impl() in the main event loop with the message and a pointer to us for the end(). 
+			///
+			/// For server stream services, in request message has already been read in. This
+			/// will then request to call Derived::impl() in the main event loop with the message
+			/// and a pointer to us. The typical implementation just checks the validity of the request,
+			/// and then adds a `std::weak_ptr`, along with some other information into a container
+			/// of listening RPC clients. When that call finishes, the work queue is started so that
+			/// outgoing messages can be written.
+			///
+			/// For bidirectional streams, message has not been read in. It requests Detail::impl() be
+			/// called in the main loop and starts the work queue for writes and reads. The typical
+			/// implementation reads in the message, checks validity, then adds a `std::weak_ptr` along
+			/// with other details to a container of listeners.
+			///
+			template<typename X = RPCType, boost::enable_if_t<std::is_same<X, Unary_t>::value>* = nullptr>
```

Comment:
`boost::enable_if_t` is a somewhat recent addition to Boost. Recent enough to not be part of the version packaged in the Ubuntu repos (at least not all of them).

Thus you should check which version of Boost or in this case which version of the `type_traits` module is installed and issue an error if it is not sufficient.